### DISCUSSION TITLE
[improvement] Use OS CA certs when possible

### DISCUSSION THREAD

**Discussion by captn3m0** [Kong/insomnia#3719] [Ideas]
- Insomnia Version: 6.3.2.1
- Operating System: Linux  5.0.7-arch1-1-ARC

## Details

Ref: 

https://github.com/getinsomnia/insomnia/blob/776b3bd31e86f947dac26709a81d670cb3acc416/packages/insomnia-app/app/network/network.js#L348-L362

`libcurl` on linux gives the same functionality, and there should be no need to rely on a insomnia-provided CA bundle.

**Comment by welcome** [comment]
👋 Thanks for opening your first issue! If you're reporting a 🐞 bug, please make sure
you include steps to reproduce it. If you're requesting a feature 🎁, please provide real
use cases that would benefit. 👪

To help make this a smooth process, please be sure you have first read the
[contributing guidelines](https://github.com/getinsomnia/insomnia/blob/master/CONTRIBUTING.md).

**Comment by gschier** [comment]
Do you know if this is this new curl behavior? A few years ago this logic was definitely required and was the suggested way of doing in by the libcurl docs. It's the reason they provide an up-to-date CA bundle https://curl.haxx.se/docs/caextract.html

I can't seem to find any mention of this requirement in the docs anymore but I also cannot find any information on the change. 

Also remember, Insomnia does not use the system's `libcurl` on Linux and Windows. It instead bundles its own to prevent dependency issues.

**Comment by captn3m0** [comment]
I meant that it comes bundled with every linux distro these days, and insomnia on linux should set up the ca-certificates package as a dependency instead of bundling its own.

I thought that this must be standardized in some way, but [it doesn't look like it](https://serverfault.com/a/722646).

Node seems to come with it hardcoded though: https://github.com/nodejs/node/blob/master/src/node_root_certs.h

~Wondering if that could be used instead?~ Not present on all node installations.

(I'm sure there is a NPM package somewhere that finds the correct path for system root CA certs, but I can't find it)

**Comment by gschier** [comment]
Based on that it seems like the best idea might be to check those common directories and fallback to the bundled CAs if they don't exist. We could also make the path configurable in the app preferences. 

Let me know if you want to take this one on and I can point you in the right direction (if needed).

**Comment by captn3m0** [comment]
Sounds good. I'll wrap it into a separate NPM package if that makes sense (or if I can't find one).

Just checking, but this wouldn't affect https://github.com/getinsomnia/insomnia/issues/675 right?

**Comment by gschier** [comment]
Yes, it should not affect #675. Unless you are also planning on making it configurable too. In that case, you'd be able to close #675.

**Comment by captn3m0** [comment]
Package created: https://www.npmjs.com/package/find-root-ca-cert
